### PR TITLE
Create IsAllowed and deprecate Allowed

### PR DESCRIPTION
**Pull request by janisz**
## Description

Deprecate Allowed and create IsAllowed instead to simiplify the code as we don't need to return error after removal of auth plugin.

## Testing Performed

CI

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2726** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2726" target="_blank">Graphite</a> 
    * **PR #2722** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2722" target="_blank">Graphite</a> 
      * **PR #2750** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2750" target="_blank">Graphite</a> 
        * **PR #2751** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2751" target="_blank">Graphite</a> 
          * **PR #2752** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2752" target="_blank">Graphite</a> 
            * **PR #2774** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2774" target="_blank">Graphite</a> 
              * **PR #2775** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2775" target="_blank">Graphite</a> 
                * **PR #2776** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2776" target="_blank">Graphite</a> 
                  * **PR #2777** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2777" target="_blank">Graphite</a> 
                    * **PR #2778** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2778" target="_blank">Graphite</a> 
                      * **PR #2779** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2779" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2779?utm_source=stack-comment).

File: pkg/sac/tests/scope_checker_or_test.go [hunk 1]
```diff
@@ -35,29 +34,51 @@ func (suite *orScopeCheckerTestSuite) SetupTest() {
...
-	suite.scopeChecker2.EXPECT().Allowed(gomock.Any()).Return(false, nil)
+	suite.scopeChecker1.EXPECT().IsAllowed(gomock.Any()).Return(false)
+	suite.scopeChecker2.EXPECT().IsAllowed(gomock.Any()).Return(false)
 
 	allowed, err = suite.orScopeChecker.Allowed()
 	suite.False(allowed)
 	suite.NoError(err)
 	suite.Nil(err)
 
 	// 3. Expect an error and False when all ScopeCheckers return Deny and at least 1 returned an error.
```
Comment (rhybrillou) [inline pkg/sac/tests/scope_checker_or_test.go:53]: There is no more error returned. This part of the test can be removed.

File: pkg/sac/tests/scope_checker_or_test.go [hunk 2]
```diff
@@ -35,29 +34,51 @@ func (suite *orScopeCheckerTestSuite) SetupTest() {
...
+	suite.True(allowed)
+
+	// 2. Expect False when all ScopeCheckers return Deny.
+	suite.scopeChecker1.EXPECT().IsAllowed(gomock.Any()).Return(false)
+	suite.scopeChecker2.EXPECT().IsAllowed(gomock.Any()).Return(false)
+
+	allowed = suite.orScopeChecker.IsAllowed()
+	suite.False(allowed)
+
+	// 3. Expect an error and False when all ScopeCheckers return Deny and at least 1 returned an error.
```
Comment (rhybrillou) [inline pkg/sac/tests/scope_checker_or_test.go:76]: The tested function does not return any error. This part can be removed.

**Review by rhybrillou — APPROVED** [review]

**Comment by janisz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2779?utm_source=stack-comment).


Job Reference: KaXF6lRV7xj8fvvZbgOl

**Pull request merged**